pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -183,21 +183,26 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		return reconcile.Result{}, errors.Wrap(err, "failed to populate cluster info")
 	}
 	r.clusterInfo.Context = r.opManagerContext
-	cephversion, err := detectCephVersion(
-		r.opManagerContext,
-		r.opConfig.Image,
-		cephBlockPoolRadosNamespace.Namespace,
-		controllerName,
-		k8sutil.NewOwnerInfo(cephBlockPoolRadosNamespace, r.scheme),
-		r.context.Clientset,
-		&cephCluster.Spec,
-	)
-	if err != nil {
-		return reconcile.Result{}, errors.Wrap(err, "failed to detect ceph version")
-	}
-	if cephversion != nil {
-		r.clusterInfo.CephVersion = *cephversion
+	// If no image is specified we don't perform any checks,
+	// for external clusters we don't need to check the version
+	if cephCluster.Spec.CephVersion.Image != "" {
+		cephversion, err := detectCephVersion(
```

Comment:
Remind me, why do we need to detect the ceph version for every reconcile of the rados namespace?

Reply:
If we really need to detect the version, can we read the CephCluster status to get the info? That would be a lot quicker and less complex than running the version job.